src/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -626,4 +682,32 @@ private void sendBroadcastNewDownload(DownloadFileOperation download, String lin
         sendStickyBroadcast(added);
     }
 
+    /**
+     * Remove downloads of an account
+     * @param account
+     */
+    private void cancelDownloadsForAccount(Account account){
```

Comment:
This code is duplicating almost all the code in```cancel(Account ...)```; replace the block there with a call here.